### PR TITLE
PXP-8168 Automatically skip tests if changes are doc-only

### PR DESCRIPTION
**Pull request by paulineribeyre**
Jira Ticket: [PXP-8168](https://ctds-planx.atlassian.net/browse/PXP-8168)

Tested here: https://github.com/uc-cdis/gen3-qa/pull/635

### Improvements
- Automatically skip tests if changes are doc-only, instead of relying on a label

### CODE REVIEW

File: vars/microservicePipeline.groovy [hunk 1]
```diff
@@ -35,6 +35,10 @@ def call(Map config) {
        try {
         // giving a chance for auto-label gh actions to catch up
         sleep(30)
+
+        // if the changes are doc-only, automatically skip the tests
+        doNotRunTests = doNotRunTests || docOnlyHelper.checkTestSkippingCriteria()
```
Comment (williamhaley) [inline vars/microservicePipeline.groovy:40]: I probably don't understand our pipeline enough, but I wonder if we should just be aborting early whenever `doNotRunTests == true` 🤔 Not sure why we'd keep running through things if we know we're not going to run tests. Maybe a bigger change than we want to take on right now 😄

File: vars/microservicePipeline.groovy [hunk 2]
```diff
@@ -45,17 +49,13 @@ def call(Map config) {
               selectedTest = "suites/" + selectedTestLabel[1] + "/" + selectedTestLabel[2] + ".js"
               selectedTests.add(selectedTest)
               break
-            case "doc-only":
```
Comment (williamhaley) [inline vars/microservicePipeline.groovy:48]: I'm guessing this is a GH label? Do we not want to use that anymore? Should we remove it from GH?

Comment (williamhaley) [reply vars/microservicePipeline.groovy:48]: (not even sure if that makes sense, but just checking)

**Review by williamhaley — APPROVED** [review]
Overall lgtm! Just a couple questions

**Review by jingh8 — APPROVED** [review]
+1

File: vars/microservicePipeline.groovy (hunk 1 above)
Comment (paulineribeyre) [reply vars/microservicePipeline.groovy:40]: yeah, i agree but i do think that's not in scope of this PR

File: vars/microservicePipeline.groovy (hunk 2 above)
Comment (paulineribeyre) [reply vars/microservicePipeline.groovy:48]: we can remove it from all the repos if we want to 🤷‍♀️  we won't need it anymore since it's automatic now

Comment (williamhaley) [reply vars/microservicePipeline.groovy:48]: Not worth the effort really. I was just thinking about any confusion if people relabel things and expect something to happen but it doesn't. I don't know if that's a thing that would actually happen IRL though. Probably not a real issue.

Comment (paulineribeyre) [reply vars/microservicePipeline.groovy:48]: if the PR is really doc-only, tests will be skipped regardless of whether the label was added, so it doesn't break anything

Comment (paulineribeyre) [reply vars/microservicePipeline.groovy:48]: if it's not doc-only, tests would not be skipped even if people added the label, before or after this change

**Pull request merged**